Sample Applications/WPFGallery/ViewModels/Samples/UserDashboardPageViewModel.cs

```diff
@@ -21,52 +21,104 @@ public partial class UserDashboardPageViewModel : ObservableObject
         [ObservableProperty]
         private User? _editableUser;
 
+        [ObservableProperty]
+        private bool _isRead = true;
+
+        [ObservableProperty]
+        private bool isSaved;
+
+        [ObservableProperty]
+        private bool isDeleted=false;
+
+        [ObservableProperty]
+        private string _deletedname;
         partial void OnSelectedUserChanged(User? oldValue, User? newValue)
         {
-            EditableUser = null;
+            if (SelectedUser == null)
+            {
+                
+            }
+            else if (SelectedUser != EditableUser)
+            {
+                EditableUser = new User(SelectedUser);
+                IsRead = true;
+                IsEditing = false;
+            }
         }
 
         [RelayCommand]
         private void AddUser()
-        { 
+        {
             Users.Add(new User("New User", ""));
             SelectedUser = Users.Last();
             EditableUser = new User(SelectedUser);
+            IsRead = false;
+            IsEditing = true;
         }
 
         [RelayCommand]
         private void RemoveUser(object selectedUser)
         {
-            if(selectedUser is User user)
+            if (selectedUser is User user)
             {
+        
+                Deletedname = user.Name;
+                IsDeleted = true;
+
+                Task.Delay(2000).ContinueWith(_ => IsDeleted = false, TaskScheduler.FromCurrentSynchronizationContext());
+                int index = Users.IndexOf(user);
+                
+                SelectedUser = Users[index+1];
                 Users.Remove(user);
-                SelectedUser = null;
+                IsRead = true;
+                IsEditing = false;
+
+                
             }
         }
 
         [RelayCommand]
         private void EditUserStart()
         {
-            EditableUser = new User(SelectedUser);
+            
+            if (SelectedUser != null)
+            {
+                
+                IsRead = false;
+                IsEditing = true;
```

Comment:
We can combine IsRead and IsEditing using an InvertedBooleanConverter ( check this : https://stackoverflow.com/questions/1039636/how-to-bind-inverse-boolean-properties-in-wpf )